### PR TITLE
sendgrid-teammates: add logic to delete users

### PR DESCRIPTION
**Pull request by jmelis**

### CODE REVIEW

File: reconcile/sendgrid_teammates.py [hunk 1]
```diff
@@ -47,28 +55,52 @@ def fetch_current_state(sg_client):
...
             LOG.info(['delete', user.email])
+            if not dry_run:
+                if user.pending:
+                    delete_method = sg_client.teammates.pending
+                    identifier = user.pending_token
+                else:
+                    delete_method = sg_client.teammates
+                    identifier = user.email
+
+                response = delete_method._(identifier).delete()
```
Comment (jmelis) [inline reconcile/sendgrid_teammates.py:96]: This underscore thing is tricky. I kind of inferred it from here: https://github.com/sendgrid/sendgrid-python#general-v3-web-api-usage-without-fluent-interface

File: reconcile/sendgrid_teammates.py [hunk 2]
```diff
@@ -47,28 +55,52 @@ def fetch_current_state(sg_client):
...
         state.append(t)
 
     return state
 
 
+def raise_if_error(response):
+    """
+    Raises an SendGridAPIError if the request has returned an error
+    """
+    if int(response.status_code / 100) != 2:
```
Comment (apahim) [inline reconcile/sendgrid_teammates.py:68]: `if response.status_code >= 300:` is 3x faster :)

File: reconcile/sendgrid_teammates.py (hunk 1 above)
Comment (apahim) [inline reconcile/sendgrid_teammates.py:96]: that's interesting... first time I see the `_` used that way.

Comment (jmelis) [reply reconcile/sendgrid_teammates.py:96]: me too, quite ingenious, though

**Review by apahim — APPROVED** [review]
lgtm

File: reconcile/sendgrid_teammates.py (hunk 2 above)
Comment (jmelis) [reply reconcile/sendgrid_teammates.py]: good idea, yes :D

**Pull request merged**